### PR TITLE
[timeseries] Update direct forecasting model based on TabularPredictor

### PR DESCRIPTION
**Pull request by shchur**
*Description of changes:*
- Rename the model `AutoGluonTabularModel` -> `DirectTabularModel` to differentiate it from `RecursiveTabularModel` also based on the tabular predictor
- Use `problem_type="quantile"` inside TabularPredictor if an eval_metric based quantiles is provided.
- Do not use the external validation set as tuning data to avoid overfitting & reporting overly optimistic val score.
- Fix a bug where future values of `known_covariates` were ignored by the model, effectively treating all covariates as past-only.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/tutorials/timeseries/forecasting-indepth.ipynb [hunk 1]
```diff
@@ -354,7 +354,7 @@
     "- `SimpleFeedForward`\n",
     "- `TemporalFusionTransformer`\n",
     "\n",
-    "AutoGluon also offers a tree-based global model `AutoGluonTabular`.\n",
+    "AutoGluon also offers a tree-based global model `DirectTabular`.\n",
```
[the remark this answers is not in the document]
Comment (shchur) [reply docs/tutorials/timeseries/forecasting-indepth.ipynb:357]: I think that "direct" is a somewhat established term for such models, e.g. https://forecastegy.com/posts/multiple-time-series-forecasting-with-scikit-learn/#recursive-vs-direct-forecasting, or [this survey paper](https://arxiv.org/pdf/2202.01712.pdf)

image

We are not 100% following the above methodology - we train a single model that forecasts all timesteps instead of single model per timestep (otherwise would be computationally infeasible for long horizons).

----
Do you think it would make sense to use "NonRecursive" in this case to differentiate from the above strategies, or are you fine with "Direct" as we are close enough to what people mean by this term?

**Review by tonyhoo — APPROVED** [review]
